sound/soc/sof/ipc4-topology.c

```diff
@@ -965,7 +965,7 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 				   struct snd_pcm_hw_params *params,
 				   struct sof_ipc4_available_audio_format *available_fmt)
 {
-	struct sof_ipc4_pin_format *pin_fmt = available_fmt->ref_audio_fmt;
+	struct sof_ipc4_pin_format *pin_fmt = available_fmt->source_audio_fmts;
```

Comment:
source is also misleading IMHO, we had source and sink pins.

Maybe use 'current' or 'selected' to show that it's either input or output pointers depending on direction.

or 'direction_format' maybe?